packages/paste-website/src/components/site-wrapper/site-header/index.tsx

```diff
@@ -0,0 +1,25 @@
+import * as React from 'react';
+import {SiteHeaderDesktop} from './SiteHeaderDesktop';
+import {SiteHeaderMobile} from './SiteHeaderMobile';
+import {useWindowSize} from '../../../hooks/useWindowSize';
+
+export const SiteHeader: React.FC = () => {
+  const [value, setValue] = React.useState('');
+  const {breakpointIndex} = useWindowSize();
+
+  // While SSR, render both and let CSS handle it (fixes FoUC)
+  if (breakpointIndex === undefined) {
```

Comment:
I'm unsure what value this technique brings right now. For the SSR payload you're having to render both anyway. From that point, I don't see what the benefit of adding JavaScript to take over something that CSS is already handling with the responsive display props.

I think we should just use what the system provides

Reply:
Initially I was only using the responsive style props to change the display css property as you mentioned. As I began building the responsive UI, I thought it would be cleaner in both the code and the UX to share the navigation state rather than have a unique copy for desktop and mobile.  

When I started sharing the state and working on the mobile header's navigation animation, I wasn't getting smooth 60fps animations. In my efforts to improve the performance of the animation, I was investigating what React was doing that could cause lag.  Turns out, even though the content was hidden in CSS, it was still being (re)rendered in the DOM.  To cancel out all that work, I figured it would be best not to render it at all.  That's when you called out the FoUC.  I then came up with this approach which would render both desktop and mobile in the static SSR copy (which is fine), and then conditionally render whatever actually needs rendering.

This isn't really a necessary change, but it helps the UX by improving the performance a tad. I don't think I'd use this everywhere, and you rightfully called out a place where I used it instead of a responsive prop. In the case of this large DOM tree of a site-header, I thought it best to keep it.